pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -11,10 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from abc import ABC, abstractmethod
+from abc import abstractmethod
 
+from pytorch_lightning.plugins.base_plugin import Plugin
 
-class ClusterEnvironment(ABC):
+
+class ClusterEnvironment(Plugin):
```

Comment:
there's no intersection between the plugin interface and cluster environment so this extension doesn't feel right to me: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/plugins/base_plugin.py

maybe we could have the `plugins` argument take a Union for Plugin and ClusterEnvironment? 

or we could just have cluster environment be its own trainer argument, similar to the training type discussion?